kani-compiler/src/codegen_cprover_gotoc/overrides/hooks.rs

```diff
@@ -317,6 +317,64 @@ impl<'tcx> GotocHook<'tcx> for SliceFromRawPart {
     }
 }
 
+/// This hook intercepts calls to `memcmp` and skips CBMC's pointer checks if the number of bytes to be compared is zero.
+/// See issue https://github.com/model-checking/kani/issues/1489
+pub struct MemCmp;
+
+impl<'tcx> GotocHook<'tcx> for MemCmp {
+    fn hook_applies(&self, tcx: TyCtxt<'tcx>, instance: Instance<'tcx>) -> bool {
+        let name = with_no_trimmed_paths!(tcx.def_path_str(instance.def_id()));
+        name.contains("memcmp")
```

Comment:
I'm wondering if this check is too open...

Reply:
Yes, you were right :D It made the tests fail. I fixed it to match the exact name (`core::slice::cmp::memcmp`).

Reply:
We used to have hooks for a bunch of memory operations. I do remember they had to also take into consideration the top module being named `std::`. @danielsn might have better insight why that was the case.

Reply:
The modules get re-exported. So we often see it either as "std" or "core", depending on whether it was re-exported.